Gateway/Request/CaptureDataBuilder.php

```diff
@@ -100,19 +108,21 @@ public function build(array $buildSubject)
     {
         /** @var \Magento\Payment\Gateway\Data\PaymentDataObject $paymentDataObject */
         $paymentDataObject = \Magento\Payment\Gateway\Helper\SubjectReader::readPayment($buildSubject);
-        $amount = \Magento\Payment\Gateway\Helper\SubjectReader::readAmount($buildSubject);
-
         $payment = $paymentDataObject->getPayment();
         /** @var Order $order */
         $order = $payment->getOrder();
+        /** @var \Magento\Sales\Model\Order\Invoice $latestInvoice */
+        $latestInvoice = $order->getInvoiceCollection()->getLastItem();
+        $invoiceAmountCurrency = $this->chargedCurrency->getInvoiceAmountCurrency($latestInvoice);
+        $currency = $invoiceAmountCurrency->getCurrencyCode();
+        $amount = $this->adyenHelper->formatAmount($invoiceAmountCurrency->getAmount(), $currency);
+        $orderAmountCents = $this->adyenHelper->formatAmount($order->getGrandTotal(), $currency);
```

Comment:
I think that here it is possible that `$order->getGrandTotal()` won’t match `$currency`, since the latter was fetched with the charged currency helper

Reply:
That may be possible, but the issue with using the `ChargedCurrency` helper here is that we would need to use the `getOrderAmountCurrency` function, which in turn utilizes the `charged_currency` config value which is not on v6.